pkg/controller/engine.go

```diff
@@ -160,6 +161,13 @@ func For(kind client.Object, h handler.EventHandler, p ...predicate.Predicate) W
 	return Watch{kind: kind, handler: h, predicates: p}
 }
 
+// TriggeredBy returns a custom watch for secondary resources triggering the
+// controller. Events will be handled by the supplied EventHandler, and may be
+// filtered by the supplied predicates.
+func TriggeredBy(cache cache.Cache, kind client.Object, h handler.EventHandler, p ...predicate.Predicate) Watch {
```

Comment:
I assume it was tested e2e by https://github.com/crossplane/crossplane/pull/4582/files#diff-52e214a3e1aa3741fcbc3dd3faca4e6ae54e98b177690300591a4d147294a70dR416 ? 

Do we want to extend unit test at https://github.com/crossplane/crossplane-runtime/blob/master/pkg/controller/engine_test.go ?

Reply:
The engine is basically untested today. It only checks some errors.

Reply:
ok, so maybe we should separately track extending unit test coverage? I'll leave it up to you and not hold this one :)

Reply:
I tried now for 20min. You mock yourself to death with cache, informer, controller. At the end, you are not sure anymore what you are testing. There is a reason this is only lightly tested 🤷

Reply:
@sttts got it, thanks a lot for checking! 👍

Reply:
Yeah, this whole package is kinda just syntactic sugar. 😬